bin/linkfix.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+if [ "$1" == "-h" -o "$1" == "--help" ]; then
+    echo "$0 [-h] REPO"
+    echo "Normalizes the symlinks in the raw subdirectory of the repo"
+    echo "by linking to the ultimate destination."
+    echo "With -h, uses hard links instead of symlinks, allowing the"
+    echo "originals to be removed if desired."
+    exit 0
+fi
+symlink="-s"
+if [ "$1" == "-h" ]; then
+    symlink=""
+    shift
+fi
+
+find "$1/raw" -type l -print | \
+    while read link; do
+        src=`readlink "$link"`
+        while [ -L "$src" ]; do
+            src=`readlink "$src"`
+        done
+        rm "$link"
+        ln $symlink "$src" "$link"
+    done
```

Comment:
I'm not really sure what this was supposed to do, and would like some clarification to what I was doing incorrectly.

 I ran this script on the DESTDIR directory after running the "reingest.sh" script based on what the "reinvest.sh" script said.  It wasn't apparent what "linkfix.sh" did.  The links didn't change with the default.   When I changed this script to remove the "-h" on line 3 to try the hard-linking feature, I got the message "ln: failed to create hard link 'butler2/raw/2020-01-28/2020012800001-det000.fits' => '/data/tmp/butler/repo/raw/2020-01-28/4617759365108394614-det000.fits': Operation not permitted" repeated.

Reply:
If the files in the source repo raw directory are already actual files (because of ingestion with `mode=move`, then the `linkfix.sh` script will not do anything in symbolic link mode.  In hard link mode, it will change the symbolic links created with `reingest.sh` into hard links.

If the files in the source repo raw directory are actually symlinks (as they are at NCSA in `/lsstdata/offliine/teststand/auxTel/gen2repo` or `/project/shared/auxTel`), then `linkfix.sh` will re-point them from the source repo to the original actual files, either as symlinks or hard links.

Of course hard links can only be created if the files are on the same filesystem.  I didn't think it was necessary to check for this, but I will add more documentation and an error message so that this is explained and so that failures die immediately.

Reply:
Thanks for the explanation, and fix